pkg/backends/otlp/internal/data/event.go

```diff
@@ -0,0 +1,125 @@
+package data
+
+import (
+	"errors"
+	"time"
+
+	v1common "go.opentelemetry.io/proto/otlp/common/v1"
+	v1log "go.opentelemetry.io/proto/otlp/logs/v1"
+
+	"github.com/atlassian/gostatsd"
+)
+
+type OtlpEvent struct {
+	raw               *gostatsd.Event
+	titleAttrKey      string
+	propertiesAttrKey string
+}
+
+func NewOtlpEvent(e *gostatsd.Event, opts ...Option) (*OtlpEvent, error) {
+	if e == nil {
+		return nil, errors.New("event not found")
+	}
+	oe := &OtlpEvent{raw: e}
+	for _, opt := range opts {
+		opt(oe)
+	}
+	return oe, nil
+}
+
+type Option func(*OtlpEvent)
+
+func WithTitleAttrKey(key string) func(*OtlpEvent) {
+	return func(e *OtlpEvent) {
+		e.titleAttrKey = key
+	}
+}
+
+func WithPropertiesAttrKey(key string) func(*OtlpEvent) {
+	return func(e *OtlpEvent) {
+		e.propertiesAttrKey = key
+	}
+}
+
+func (s *OtlpEvent) TransformToLog() *v1log.LogRecord {
+	attrs := eventLogAttributes(make([]*eventLogAttribute, 0))
+	e := s.raw
+	dimensions := e.Tags.ToMap(string(e.Source))
```

Comment:
Hmm why don't we just straight up put `source` in attrs rather than putting it to `tags` then `dimensions` then `attrs`?

Reply:
good question, the logic was the source is converted into `host` in `toMap`.  It should be keep as it is and be converted as needed in the downstream services.